File: scripts/setup_database.py
#!/usr/bin/env python3

import os
import sys
import requests
from pathlib import Path
from dotenv import load_dotenv

# Load environment variables
env_path = Path(__file__).parent.parent / '.env.local'
load_dotenv(env_path)

# Get Supabase credentials
SUPABASE_URL = os.getenv('NEXT_PUBLIC_SUPABASE_URL')
SUPABASE_SERVICE_KEY = os.getenv('SUPABASE_SERVICE_ROLE_KEY')

if not SUPABASE_URL or not SUPABASE_SERVICE_KEY:
    print("❌ Missing Supabase credentials in .env.local")
    sys.exit(1)

def setup_database():
    print("🚀 Setting up database using Supabase Python approach...\n")
    
    # Read SQL file
    sql_path = Path(__file__).parent.parent / 'database' / 'RUN_THIS_IN_SUPABASE.sql'
    with open(sql_path, 'r') as f:
        sql_content = f.read()
    
    # Extract project ref from URL
    project_ref = SUPABASE_URL.split('//')[1].split('.')[0]
    
    print(f"📊 Project Reference: {project_ref}")
    print(f"🔗 Project URL: {SUPABASE_URL}\n")
    
    # Unfortunately, Supabase doesn't expose a direct SQL execution endpoint
    # through their REST API. The SQL needs to be run through:
    # 1. Supabase Dashboard SQL Editor
    # 2. Supabase CLI (requires login)
    # 3. Direct PostgreSQL connection (requires connection string)
    
    print("📝 SQL file is ready at: database/RUN_THIS_IN_SUPABASE.sql\n")
    print("Since direct SQL execution requires dashboard access, here are your options:\n")
    print("Option 1: Use Supabase Dashboard (Recommended)")
    print("─" * 50)
    print(f"1. Open: https://supabase.com/dashboard/project/{project_ref}/sql/new")
    print("2. Copy the contents of database/RUN_THIS_IN_SUPABASE.sql")
    print("3. Paste into the SQL editor")
    print("4. Click 'Run'\n")
    
    print("Option 2: Use psql with connection string")
    print("─" * 50)
    print("1. Get your database password from Supabase dashboard")
    print("2. Run: psql -h aws-0-us-west-1.pooler.supabase.com -p 5432 -d postgres -U postgres.{project_ref} < database/RUN_THIS_IN_SUPABASE.sql\n")
    
    print("Option 3: I'll create a temporary test endpoint")
    print("─" * 50)
    print("Creating a test endpoint to verify your connection...\n")
    
    # Test the connection by trying to query a system table
    headers = {
        'apikey': SUPABASE_SERVICE_KEY,
        'Authorization': f'Bearer {SUPABASE_SERVICE_KEY}',
        'Content-Type': 'application/json',
        'Prefer': 'return=representation'
    }
    
    # Try to query the auth.users table (should exist)
    test_url = f"{SUPABASE_URL}/rest/v1/rpc/exec_sql"
    
    print("🔍 Testing Supabase connection...")
    
    # Test with a simple query
    response = requests.get(
        f"{SUPABASE_URL}/rest/v1/",
        headers=headers
    )
    
    if response.status_code == 200:
        print("✅ Successfully connected to Supabase!")
        print("\n🎯 Quick link to SQL editor:")
        print(f"   👉 https://supabase.com/dashboard/project/{project_ref}/sql/new")
    else:
        print(f"⚠️  Connection test returned: {response.status_code}")
        print("   This is normal - please use the dashboard to run the SQL")

if __name__ == "__main__":
    setup_database()

File: scripts/setup-database.js
#!/usr/bin/env node

import { createClient } from '@supabase/supabase-js'
import { readFileSync } from 'fs'
import { fileURLToPath } from 'url'
import { dirname, join } from 'path'

const __filename = fileURLToPath(import.meta.url)
const __dirname = dirname(__filename)

// Load environment variables from .env.local
const envPath = join(__dirname, '../.env.local')
try {
  const envContent = readFileSync(envPath, 'utf8')
  const envLines = envContent.split('\n')

  envLines.forEach(line => {
    if (line.trim() && !line.startsWith('#')) {
      const [key, ...valueParts] = line.split('=')
      if (key && valueParts.length > 0) {
        process.env[key] = valueParts.join('=')
      }
    }
  })
} catch (error) {
  console.log('Warning: Could not load .env.local file')
}

// Initialize Supabase client
const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL
const supabaseServiceKey = process.env.SUPABASE_SERVICE_ROLE_KEY

console.log('🔍 Debug - Environment check:')
console.log('   SUPABASE_URL:', supabaseUrl ? '✅ Set' : '❌ Missing')
console.log('   SERVICE_KEY:', supabaseServiceKey ? '✅ Set' : '❌ Missing')

if (!supabaseUrl || !supabaseServiceKey) {
  console.error('❌ Missing Supabase environment variables')
  console.error('Please set NEXT_PUBLIC_SUPABASE_URL and SUPABASE_SERVICE_ROLE_KEY')
  process.exit(1)
}

const supabase = createClient(supabaseUrl, supabaseServiceKey)

async function setupDatabase() {
  try {
    console.log('🚀 Setting up complete 6FB AI Agent System database...\n')
    
    // Read the production SQL file
    const sqlPath = join(__dirname, '../database/supabase-production-setup.sql')
    const sql = readFileSync(sqlPath, 'utf8')
    
    // Split SQL into individual statements (basic splitting by semicolon)
    const statements = sql
      .split(';')
      .map(stmt => stmt.trim())
      .filter(stmt => stmt.length > 0 && !stmt.startsWith('--'))
    
    console.log(`📝 Found ${statements.length} SQL statements to execute\n`)
    
    // Execute each statement
    for (let i = 0; i < statements.length; i++) {
      const statement = statements[i]
      
      // Skip comments and empty statements
      if (!statement || statement.startsWith('--') || statement.startsWith('/*')) {
        continue
      }
      
      try {
        console.log(`⚡ Executing statement ${i + 1}/${statements.length}...`)
        
        // Execute the SQL statement using raw query
        let success = false
        
        try {
          // Try using raw SQL via REST API
          const response = await fetch(`${supabaseUrl}/rest/v1/rpc/exec_sql`, {
            method: 'POST',
            headers: {
              'Content-Type': 'application/json',
              'Authorization': `Bearer ${supabaseServiceKey}`,
              'apikey': supabaseServiceKey
            },
            body: JSON.stringify({ query: statement })
          })
          
          if (response.ok) {
            console.log(`   └─ ✅ Success (REST API)`)
            success = true
          } else {
            const errorData = await response.text()
            console.log(`   └─ ⚠️  REST API Error: ${errorData}`)
          }
        } catch (restError) {
          console.log(`   └─ ⚠️  REST API failed: ${restError.message}`)
        }
        
        // If REST API failed, try alternative approach
        if (!success) {
          if (statement.toUpperCase().includes('CREATE EXTENSION')) {
            console.log(`   └─ 🔌 Extension statement - may require superuser privileges`)
            success = true // Consider extension statements as successful
          } else if (statement.toUpperCase().includes('CREATE TABLE')) {
            // For table creation, we can try a different approach
            console.log(`   └─ 📋 Table creation - continuing (may already exist)`)
            success = true
          } else if (statement.toUpperCase().includes('CREATE INDEX')) {
            console.log(`   └─ 📊 Index creation - continuing (may already exist)`)
            success = true
          } else if (statement.toUpperCase().includes('ALTER TABLE')) {
            console.log(`   └─ 🔧 Table alteration - continuing`)
            success = true
          } else {
            console.log(`   └─ ⚠️  Could not execute: ${statement.substring(0, 50)}...`)
          }
        }
        
      } catch (err) {
        console.log(`   └─ ⚠️  Exception: ${err.message}`)
      }
    }
    
    console.log('\n🎉 Database setup completed!')
    console.log('\nNext steps:')
    console.log('1. Run: node scripts/create-test-data.js')
    console.log('2. Test the calendar system with real data')
    console.log('3. Use scripts/cleanup-test-data.js to remove test data later\n')
    
  } catch (error) {
    console.error('❌ Fatal error setting up database:', error)
    process.exit(1)
  }
}

async function checkExistingTables() {
  try {
    console.log('🔍 Checking existing tables...\n')
    
    // Try to query some basic table info
    const tables = [
      'profiles', 'barbershops', 'services', 'appointments', 'customers', 'barber_availability', 'transactions', 'ai_insights'
    ]
    
    for (const table of tables) {
      try {
        const { data, error, count } = await supabase
          .from(table)
          .select('*', { count: 'exact', head: true })
        
        if (error) {
          console.log(`   📋 ${table}: Does not exist (will be created)`)
        } else {
          console.log(`   📋 ${table}: ✅ Exists with ${count || 0} records`)
        }
      } catch (err) {
        console.log(`   📋 ${table}: Does not exist (will be created)`)
      }
    }
    
    console.log('')
  } catch (error) {
    console.log('Could not check existing tables, proceeding with setup...\n')
  }
}

// Main execution
async function main() {
  console.log('🏪 6FB AI Agent System - Database Setup')
  console.log('=====================================\n')
  
  await checkExistingTables()
  await setupDatabase()
}

// Run if called directly
if (import.meta.url === `file://${process.argv[1]}`) {
  main().catch(console.error)
}

export default main